File: q1700/q1700.py
class Solution:
    def countStudents(self, students: List[int], sandwiches: List[int]) -> int:

        squareStudent = sum(students) 
        circleStudent = len(students) - squareStudent
        
        for sandwich in sandwiches: 
            if sandwich == 0 and circleStudent == 0: return squareStudent
            elif sandwich == 1 and squareStudent == 0: return circleStudent

            elif sandwich == 0: circleStudent -= 1
            else: squareStudent -= 1
        
        return 0

File: q1700/q1700.java
class Solution {
    public int countStudents(int[] students, int[] sandwiches) {
        int circleStudent = 0;
        int squareStudent = 0;
        for (int student : students) {
            if (student == 0)
                circleStudent++;
            else
                squareStudent++;
        }

        for (int sandwich : sandwiches) {
            if (sandwich == 0 && circleStudent == 0)
                return squareStudent;
            if (sandwich == 1 && squareStudent == 0)
                return circleStudent;

            if (sandwich == 0)
                circleStudent--;
            else
                squareStudent--;
        }
        return 0;
    }
}